src/graph_gui.cpp

```diff
@@ -1486,11 +1486,21 @@ struct IndustryProductionGraphWindow : BaseGraphWindow {
 		this->num_on_x_axis = GRAPH_NUM_MONTHS;
 		this->num_vert_lines = GRAPH_NUM_MONTHS;
 		this->month_increment = 1;
-		this->x_values_start = ECONOMY_MONTH_MINUTES;
-		this->x_values_increment = ECONOMY_MONTH_MINUTES;
-		this->draw_dates = !TimerGameEconomy::UsingWallclockUnits();
 		this->ranges = RANGE_LABELS;
 
+		StringID widgetFooterText;
+		if (TimerGameEconomy::UsingWallclockUnits()) {
+			this->x_values_start = GRAPH_NUM_MONTHS;
+			this->x_values_increment = -1 * ECONOMY_MONTH_MINUTES;
+			this->draw_dates = false;
+			widgetFooterText = STR_GRAPH_LAST_24_MINUTES_TIME_LABEL;
+		} else {
+			this->x_values_start = ECONOMY_MONTH_MINUTES;
+			this->x_values_increment = ECONOMY_MONTH_MINUTES;
+			this->draw_dates = true;
+			widgetFooterText = STR_EMPTY;
```

Comment:
OpenTTD uses `snake_case` for local variables.

Anyhow, all this fix actually changes is the type and value of `x_values_increment` and `x_values_start`. So you only need to unconditionally change 3 lines. This `if` does not make it easier.